lib/ZwaveClient.ts

```diff
@@ -0,0 +1,3276 @@
+			}
+
+			this.sendToSocket(socketEvents.nodeUpdated, node)
+		}
+	}
+
+	/**
+	 * Used to Add a new hass device to a specific node
+	 *
+	 * @param {HassDevice} hassDevice The Hass device
+	 * @param {number} nodeId The nodeid
+	 */
+	addDevice(hassDevice: HassDevice, nodeId: number) {
+		const node = this.nodes.get(nodeId)
+
+		// check for existing node and node hassdevice with given id
+		if (node && hassDevice.id) {
+			delete hassDevice.id
+			const id = hassDevice.type + '_' + hassDevice.object_id
+			hassDevice.persistent = false
+			node.hassDevices[id] = hassDevice
+
+			this.sendToSocket(socketEvents.nodeUpdated, node)
+		}
+	}
+
+	/**
+	 * Used to update hass devices list of a specific node and store them in `nodes.json`
+	 *
+	 */
+	async storeDevices(
+		devices: { [key: string]: HassDevice },
+		nodeId: number,
+		remove: any
+	) {
+		const node = this.nodes.get(nodeId)
+
+		if (node) {
+			for (const id in devices) {
+				devices[id].persistent = !remove
+			}
+
+			if (remove) {
+				delete this.storeNodes[nodeId].hassDevices
+			} else {
+				this.storeNodes[nodeId].hassDevices = devices
+			}
+
+			node.hassDevices = utils.copy(devices)
+			await jsonStore.put(store.nodes, this.storeNodes)
+
+			this.sendToSocket(socketEvents.nodeUpdated, node)
+		}
+	}
+
+	/**
+	 * Method used to close client connection, use this before destroy
+	 *
+	 * @param {boolean} keepListeners Prevents to remove all ZwaveCLient listeners (used when restarting)
+	 * @memberof ZwaveClient
+	 */
+	async close(keepListeners = false) {
+		this.status = ZwaveClientStatus.CLOSED
+		this.closed = true
+
+		if (this.commandsTimeout) {
+			clearTimeout(this.commandsTimeout)
+			this.commandsTimeout = null
+		}
+
+		if (this.reconnectTimeout) {
+			clearTimeout(this.reconnectTimeout)
+			this.reconnectTimeout = null
+		}
+
+		if (this.healTimeout) {
+			clearTimeout(this.healTimeout)
+			this.healTimeout = null
+		}
+
+		if (this.updatesCheckTimeout) {
+			clearTimeout(this.updatesCheckTimeout)
+			this.updatesCheckTimeout = null
+		}
+
+		if (this.statelessTimeouts) {
+			for (const k in this.statelessTimeouts) {
+				clearTimeout(this.statelessTimeouts[k])
+				delete this.statelessTimeouts[k]
+			}
+		}
+
+		if (this.pollIntervals) {
+			for (const k in this.pollIntervals) {
+				clearTimeout(this.pollIntervals[k])
+				delete this.pollIntervals[k]
+			}
+		}
+
+		if (this.server) {
+			this.server.destroy()
+		}
+
+		if (this.driver) {
+			this.driverReady = false
+			await this.driver.destroy()
+		}
+
+		if (!keepListeners) {
+			this.removeAllListeners()
+		}
+
+		logger.info('Client closed')
+	}
+
+	getStatus() {
+		const status: {
+			driverReady: boolean
+			status: boolean
+			config: ZwaveConfig
+		} = {
+			driverReady: this.driverReady,
+			status: this.driverReady && !this.closed,
+			config: this.cfg,
+		}
+
+		return status
+	}
+
+	/**
+	 * Popolate node `groups`
+	 *
+	 * @param {number} nodeId Zwave node id
```

Comment:
```suggestion
	 * @param nodeId Zwave node id
```
I won't comment on this anymore.